sdcm/data_operations/data_operation.py

```diff
@@ -0,0 +1,293 @@
+
+        if not unique_partitions:
+            raise NoDataFound(
+                f"Not found data in the {self.schema_helpers.keyspace_name}.{self.schema_helpers.table_name}")
+
+        return [unique_partitions[random.randint(0, len(unique_partitions))] for _ in range(partitions_amount)]
+
+    def get_max_clustering_value_in_partition(self, where_condition: list) -> dict:
+        max_clustering_value = {}
+        if not (clustering_key := self.schema_helpers.clustering_key):
+            InfoEvent("There is no clustering key in the table. The maximum clustering value in partition can not be found",
+                      severity=Severity.NORMAL).publish()
+            return max_clustering_value
+
+        with self.schema_helpers.cluster.cql_connection_patient(self.schema_helpers.cluster.nodes[0],
+                                                                keyspace=self.schema_helpers.keyspace_name) as session:
+            session.default_consistency_level = ConsistencyLevel.QUORUM
+            cmd = f"select {clustering_key} as clustering_key from {self.schema_helpers.table_name} " \
+                  f"where %s order by {clustering_key} desc limit 1"
+            for where_clause in where_condition:
+                result = list(session.execute(cmd % where_clause, timeout=300))
+                if result:
+                    max_clustering_value[where_clause] = {"ck_name": clustering_key, "value": result[0].clustering_key}
+
+            if not result:
+                raise NoDataFound(
+                    f"Not found data for partition in the {self.schema_helpers.keyspace_name}.{self.schema_helpers.table_name}")
+
+        return max_clustering_value
+
+    @staticmethod
+    def build_where_condition(columns: list, values: list):
+        where_condition = []
+        for row_data in values:
+            # The WHERE condition is built for numeric column type
+            where_condition.append(" and ".join(f"{one[0]} = {one[1]}" for one in zip(columns, row_data)))
+
+        return where_condition
+
+    def delete_partition(self, where_condition_to_delete):
+        self.log.debug("Delete partitions: %s", where_condition_to_delete)
+        with self.schema_helpers.cluster.cql_connection_patient(self.schema_helpers.cluster.nodes[0],
+                                                                keyspace=self.schema_helpers.keyspace_name) as session:
+            for row in where_condition_to_delete:
+                query = f"delete from {self.schema_helpers.table_name} where {row}"
+                self.log.debug("Delete query: %s", query)
+                self.schema_helpers.session_execute(session=session, query=query, timeout=300)
+        self.log.debug("Partitions '%s' have been deleted", where_condition_to_delete)
+
+    def query_one_partition(self, where_condition_to_delete, session):
+        # TODO: add ALLOW FILTERING when need only?
```

Comment:
why do we need ALLOW FILTERING here ? if we pick one partition we shouldn't need it...